From d018d58ee581cf78ce7d110e80fddb1304edd324 Mon Sep 17 00:00:00 2001
From: Yaroslav Zviezda <acroreiser@gmail.com>
Date: Sun, 7 Apr 2024 21:29:45 +0300
Subject: [PATCH 5/5] Revert "Remove makefile_goal workaround for
 libwifi-hal-*"

This reverts commit ff1f7d9f66956653a02407cc1e4de76e05ec9af0.
---
 libwifi_hal/Android.bp | 66 +++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 65 insertions(+), 1 deletion(-)

diff --git a/libwifi_hal/Android.bp b/libwifi_hal/Android.bp
index 831277e..cc8157b 100644
--- a/libwifi_hal/Android.bp
+++ b/libwifi_hal/Android.bp
@@ -167,7 +167,7 @@ wifi_cc_defaults {
                 // whole_static_libs: ["libwifi-hal-mrvl"],
             },
             mediaTek: {
-                whole_static_libs: ["//vendor/mediatek/mokey:libwifi-hal-mt66xx"],
+                whole_static_libs: ["libwifi-hal-mt66xx"],
             },
             realtek: {
                 whole_static_libs: ["libwifi-hal-rtk"],
@@ -295,6 +295,38 @@ soong_config_module_type {
     ],
 }
 
+// libwifi-hal-syna
+
+wifi_cc_prebuilt_library_static {
+    name: "libwifi-hal-syna",
+    proprietary: true,
+    srcs: [":make-libwifi-hal-syna"],
+    compile_multilib: "first",
+    installable: false,
+    licenses: ["libwifi_hal_apache20_license"],
+    enabled: false,
+    soong_config_variables: {
+        board_wlan_device: {
+            synadhd: {
+                enabled: true,
+            },
+        },
+    },
+}
+
+wifi_makefile_goal {
+    name: "make-libwifi-hal-syna",
+    product_out_path: "obj/STATIC_LIBRARIES/libwifi-hal-syna_intermediates/libwifi-hal-syna.a",
+    enabled: false,
+    soong_config_variables: {
+        board_wlan_device: {
+            synadhd: {
+                enabled: true,
+            },
+        },
+    },
+}
+
 // libwifi-hal-qcom
 
 wifi_cc_prebuilt_library_static {
@@ -358,3 +390,35 @@ wifi_makefile_goal {
         },
     },
 }
+
+// libwifi-hal-mt66xx
+
+wifi_cc_prebuilt_library_static {
+    name: "libwifi-hal-mt66xx",
+    proprietary: true,
+    srcs: [":make-libwifi-hal-mt66xx"],
+    compile_multilib: "first",
+    installable: false,
+    licenses: ["libwifi_hal_apache20_license"],
+    enabled: false,
+    soong_config_variables: {
+        board_wlan_device: {
+            mediaTek: {
+                enabled: true,
+            },
+        },
+    },
+}
+
+wifi_makefile_goal {
+    name: "make-libwifi-hal-mt66xx",
+    product_out_path: "obj/STATIC_LIBRARIES/libwifi-hal-mt66xx_intermediates/libwifi-hal-mt66xx.a",
+    enabled: false,
+    soong_config_variables: {
+        board_wlan_device: {
+            mediaTek: {
+                enabled: true,
+            },
+        },
+    },
+}
-- 
2.45.2

